src/control/server/server.go

```diff
@@ -352,6 +352,13 @@ func (srv *server) addEngines(ctx context.Context) error {
 		return err
 	}
 
+	// Fail to start if transparent hugepages are enabled. DAOS SPDK use needs feature disabled.
+	if thpEnabled, err := topology.DefaultTHPDetector(srv.log).IsTHPEnabled(); err != nil {
```

Comment:
I'd suggest having the provider be part of the server struct. Then you could mock this and create a unit test.

Reply:
I don't want to bloat the server struct any more but I will add the interfaces as parameters to the prepBdevStorage function to enable better unit test coverage. done

Reply:
That works for me.